run/archiveloop

```diff
@@ -350,6 +360,8 @@ else
   mount_and_fix_errors_in_files
 
   connect_usb_drives_to_host
+  
+  check_if_usb_gadget_is_mounted
```

Comment:
in your original PR you called this function after the wait_for_archive_to_be_unreachable on line 383 below, which seems more useful and in the spirit of what you're actually trying to fix (the gadget becoming disconnected after a long period of inactivity).
As written here, you're basically checking whether the gadget is active immediately after activating it, which I suspect will always be the case.

Reply:
You're right. My original commit was wrong.